net/instaweb/rewriter/add_instrumentation.js

```diff
@@ -88,21 +88,30 @@ pagespeed.AddInstrumentation.prototype.sendBeacon = function() {
         timingApi['responseEnd'] - timingApi['responseStart']);
     url += '&dom_c=' + (timingApi['domContentLoadedEventStart'] - navStartTime);
 
-    if (window['performance']['navigation']) {
-      url += '&nt=' + window['performance']['navigation']['type'];
+    if (winPerf['navigation']) {
+      url += '&nt=' + winPerf['navigation']['type'];
     }
     var firstPaintTime = -1;
+    var paintEntries, loadTimes;
     if (timingApi['msFirstPaint']) {
       // IE.
       firstPaintTime = timingApi['msFirstPaint'];
-    } else if (window['chrome'] && window['chrome']['loadTimes']) {
+    } else if (winPerf['timeOrigin'] &&
```

Comment:
can you do the same trick here -- predeclaring vars for timeOrigin and getEntriesByType and then assigning them in the if-statement as side effects. That looks pretty readable.